index.js

```diff
@@ -1182,20 +1189,37 @@ module.exports = function (log, indexesPath) {
     return a.timestamp > b.timestamp
   }
 
-  function sortedByTimestamp(bitset, descending) {
+  function compareSeqAscending(a, b) {
+    return b.seq > a.seq
+  }
+
+  function compareSeqDescending(a, b) {
+    return a.seq > b.seq
+  }
+
+  function sortedBy(bitset, descending, sortBy) {
     updateCacheWithLog()
     const order = descending ? 'descending' : 'ascending'
+    const sortedCache = sortBy === 'arrival' ? sortedSeqCache : sortedTSCache
+    const comparer =
+      sortBy === 'arrival'
+        ? descending
+          ? compareSeqDescending
+          : compareSeqAscending
+        : descending
+        ? compareDescending
+        : compareAscending
+
     if (sortedCache[order].has(bitset)) return sortedCache[order].get(bitset)
-    const fpq = new FastPriorityQueue(
-      descending ? compareDescending : compareAscending
-    )
-    bitset.forEach((seq) => {
-      fpq.add({
-        seq,
-        timestamp: indexes['timestamp'].tarr[seq],
+    const fpq = new FastPriorityQueue(comparer)
+    fpq.heapify(
```

Comment:
If you look at the commits I refactored this. I noticed there was this heapify method that does exactly what we want so it should be faster but I couldn't detect this when put into ssb-db2 as there is a bit too much noise in results (+/- 5% up and down).

Reply:
Sweet